src/sentry/digests/notifications.py

```diff
@@ -32,22 +32,22 @@ class DigestInfo(NamedTuple):
 
 def split_key(
     key: str,
-) -> tuple[Project, ActionTargetType, str | None, FallthroughChoiceType | None]:
+) -> tuple[Project, ActionTargetType, int | None, FallthroughChoiceType | None]:
     key_parts = key.split(":", 5)
     project_id = key_parts[2]
     # XXX: We transitioned to new style keys (len == 5) a while ago on
     # sentry.io. But self-hosted users might transition at any time, so we need
     # to keep this transition code around for a while, maybe indefinitely.
     if len(key_parts) == 6:
         target_type = ActionTargetType(key_parts[3])
-        target_identifier = key_parts[4] if key_parts[4] else None
+        target_identifier = int(key_parts[4]) if key_parts[4] else None
```

Comment:
i think target_identifier was mistakenly identified as a `str` in some of these methods, as everywhere else (ie. all the notification classes), it is an `int`